gen.go

```diff
@@ -0,0 +1,152 @@
+package fastjson
+
+import (
+	"strconv"
+)
+
+// NewObject returns a new Value with the parameter as its initial content.
```

Comment:
It should be mentioned in the docs that the returned object is valid only until the next call to `Parser.Parse`.
The same applies to all the other `Parser.New*` functions.

Reply:
Also I'm skeptical about `Parser.New*` functions, since:
- It is easy to shoot in the foot by calling `Parser.Parse` on the `Parser` used for creating new objects.
- Continuous calls to `Parser.New*` on the same parser would lead to memory leak - the parser would hold more and more objects.

I'd just leave `New*` functions without tying them to the parser and for each `New*` function would add the corresponding `Put*`, which could be used for returning the object to a pool in order to reduce memory allocations.

Reply:
The reason for `Parser.New*` functions is that it feels more consistent with existing APIs. You use one parser for parsing a single object, so for creating a single object. Maybe let the user do a `Parser.Reset()` before creating values, just like how the user parses an object before consuming it.

Reply:
BTW I don't think using a Pool would help in this case. Value is a fixed-size object and it doesn't contain any interesting references (it's unlikely that the array/kv slices will be reused). Go's built-in memory manager and GC handles it well.